src/bsl/scopeProvider.ts

```
@@ -15,15 +15,37 @@ const scopeProvider = {
+        } else {
+            return scope.findMember(symbol.name)
         }
+    },
+
+    async resolveSymbolParentScope(model: ModelOrScope, symbol: AccessSequenceSymbol | NamedSymbol): ResolvedScope {
+        let scope: Scope | undefined = isModel(model) ? EditorScope.getScope(model) : model
 
-        return scope?.findMember(symbol.name)
+        if (symbol instanceof AccessSequenceSymbol) {
+            const lastSymbol = symbol.unclosed ? symbol.access[symbol.access.length - 1] : symbol.access[symbol.access.length - 2]
+            if (lastSymbol && lastSymbol.type) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)